types/request.d.ts

```diff
@@ -14,6 +15,11 @@ export interface RequestGenericInterface {
   Headers?: RequestHeadersDefault;
 }
 
+export interface ValidationFunction {
+  (input: any): boolean
+  errors?: null | ErrorObject[];
```

Comment:
Shall we have also an `unknown` type for allowing inference maybe?
Just to match it with my prior point that devs can implement its own compiler which bifurcates from the default behaviour?

Reply:
it seems to me that a separate issue is needed here to account for this factor

In any case, I will not be able to handle this case (